lib/msf/core/post/common.rb

```diff
@@ -232,14 +232,29 @@ def get_processes
       # adding manually because this is common for all windows I think and splitting for this was causing problem for other processes.
       processes.prepend({ 'name' => '[System Process]', 'pid' => 0 })
     else
-      ps_aux = cmd_exec('ps aux').split("\n")
-      ps_aux.delete_at(0)
-      ps_aux.each do |p|
-        properties = p.split
-        process = {}
-        process['name'] = properties[10].gsub(/\[|\]/,"")
-        process['pid'] = properties[1].to_i
-        processes.push(process)
+      if command_exists?('ps')
+        ps_aux = cmd_exec('ps aux').split("\n")
+        ps_aux.delete_at(0)
+        ps_aux.each do |p|
+          properties = p.split
+          process = {}
+          process['name'] = properties[10].gsub(/\[|\]/,"")
+          process['pid'] = properties[1].to_i
+          processes.push(process)
+        end
+      else
+        dir_proc = "/proc/"
```

Comment:
Let's check that this exists; guard against `nil` in each statement below.

Reply:
okay, will do that. thanks!

Reply:
It's a good idea to check that `/proc` exists but it should also handle cases where PIDs "vanish" during the course of iteration. This is listing the pids once and then iterating through them all, and it's possible that in a production environment the list may become stale if the connection is slow. A process list is a snapshot in time so I don't think it's a big deal if some entries are missing once all the information has been gathered, but if one process exits while this is iterating, we need to make sure that the whole thing won't fail.

Reply:
Okay, I'll add some checks to avoid that 👍🏻

Reply:
I've added a check if `/proc` exists and added checks in case some PIDs might vanish.